wavenet/ops.py

```diff
@@ -1,6 +1,87 @@
 from __future__ import division
-
+import librosa
+import numpy as np
 import tensorflow as tf
+from .audio_reader import (AudioReader, trim_silence)
+from .text_reader import TextReader
+from .image_reader import ImageReader
+from PIL import Image
+
+
+def FileReader(data_dir, coord, sample_rate, sample_size,
+               silence_threshold, quantization_channels,
+               pattern, EPSILON=0.001, raw_type="Audio"):
+    if raw_type == "Audio":
+        # Allow silence trimming to be skipped by specifying a
+        # threshold near zero.
+        silence_threshold = silence_threshold if silence_threshold > \
+                                                      EPSILON else None
+        reader = AudioReader(data_dir, coord, sample_rate=sample_rate,
+                             sample_size=sample_size,
+                             silence_threshold=silence_threshold,
+                             quantization_channels=quantization_channels,
+                             pattern=pattern)
+    elif raw_type == "Text":
+        reader = TextReader(data_dir, coord,
+                            sample_size=sample_size,
+                            pattern=pattern)
+    elif raw_type == "Image":
+        reader = ImageReader(data_dir, coord,
+                             sample_size=sample_size,
+                             pattern=pattern)
+    return reader
+
+
+def write_output(waveform, filename, sample_rate, raw_type="Audio"):
+    if raw_type == "Image":
```

Comment:
Taking the above idea a  bit further, we could have  a dictionary of FileHandler objects with the factory method for creating the right kind of Reader, a write method, bool IsAudio() and so on. That would allow us to eliminate the if statements with magic string literals that are litered across the code. I think it will be more maintainable because it won't force anyone changing the code to know all the places where those string literals appear, and the code would be more compact. For example, this if-else statement could be replaced with a single line
```
 file_handler[raw_type].write_output(waveform, filename)
```